packages/rum/src/userActionCollection.ts

```diff
@@ -66,55 +67,110 @@ export interface AutoUserAction {
+  subscriptions.push(lifeCycle.subscribe(LifeCycleEventType.VIEW_COLLECTED, processViewLoading))
+  function processViewLoading(loadedView: View) {
+    if (!loadedView || loadedView.id === currentViewId) {
+      return
+    }
+    currentViewId = loadedView.id
+    newViewLoading(lifeCycle, loadedView.location.pathname, loadedView.startTime)
+  }
 
   return {
     stop() {
       removeEventListener(DOM_EVENT.CLICK, processClick, { capture: true })
+      subscriptions.forEach((s) => s.unsubscribe())
     },
   }
 }
 
-let currentUserAction: { id: string; startTime: number } | undefined
+interface PartialUserAction {
+  id: string
+  startTime: number
+  type: UserActionType
+  name: string
+}
+let currentUserAction: PartialUserAction | undefined
+
+interface ViewLoadingState {
+  pathname: string
+  startTime: number
+  stopPageActivitiesTracking: () => void
+}
+let currentViewLoadingState: ViewLoadingState | undefined
+
+function newViewLoading(lifeCycle: LifeCycle, pathname: string, startTime: number) {
+  // Cancel current user action
+  currentUserAction = undefined
+
+  if (currentViewLoadingState) {
+    currentViewLoadingState.stopPageActivitiesTracking()
+  }
+
+  const { observable: pageActivitiesObservable, stop: stopPageActivitiesTracking } = trackPageActivities(lifeCycle)
+  currentViewLoadingState = {
+    pathname,
+    startTime,
+    stopPageActivitiesTracking,
+  }
+
+  waitUserActionCompletion(pageActivitiesObservable, (endTime) => {
+    stopPageActivitiesTracking()
+    if (currentViewLoadingState !== undefined) {
+      // Validation timeout completion does not return an end time
+      const loadingEndTime = endTime || performance.now()
+      lifeCycle.notify(LifeCycleEventType.VIEW_LOAD_COMPLETED, {
+        duration: loadingEndTime - currentViewLoadingState.startTime,
+        startTime: currentViewLoadingState.startTime,
+      })
+    }
+    currentViewLoadingState = undefined
+  })
+}
 
 function newUserAction(lifeCycle: LifeCycle, type: UserActionType, name: string) {
-  if (currentUserAction) {
-    // Discard any new user action if another one is already occuring.
+  if (currentViewLoadingState || currentUserAction) {
+    // Discard any new click user action if another one is already occuring.
+    // Discard any new user action if page is in a loading state.
     return
   }
 
   const id = generateUUID()
   const startTime = performance.now()
-  currentUserAction = { id, startTime }
+  currentUserAction = { id, startTime, type, name }
 
   const { observable: pageActivitiesObservable, stop: stopPageActivitiesTracking } = trackPageActivities(lifeCycle)
   const { eventCounts, stop: stopEventCountsTracking } = trackEventCounts(lifeCycle)
 
   waitUserActionCompletion(pageActivitiesObservable, (endTime) => {
     stopPageActivitiesTracking()
     stopEventCountsTracking()
-    if (endTime !== undefined) {
+    if (endTime !== undefined && currentUserAction !== undefined) {
```

Comment:
currentUserAction was supposed to be set to undefined before reaching this point by waitForUserActionCompletion.
Could it be done before this check and could we remove this undefined check?
If not, there is a test case missing

Reply:
I can remove the `currentUserAction !== undefined`from the updated version of the call but, as we are a new 'waitIdlePageActivity' module that replace `waitForUserActionCompletion` we implement that 'currentUserAction was supposed to be set to undefined before reaching this point by waitForUserActionCompletion' anymore